src/Blocks/BlockPiston.cpp

```diff
@@ -86,28 +73,129 @@ bool cBlockPistonHandler::GetPlacementBlockTypeMeta(
 
 
 
-int cBlockPistonHandler::FirstPassthroughBlock(int a_PistonX, int a_PistonY, int a_PistonZ, NIBBLETYPE pistonmeta, cWorld * a_World)
+Vector3i cBlockPistonHandler::VectorFromMetaData(int a_PistonMeta)
```

Comment:
The function should be `static`, because it doesn't depend on any state of the instance. No need to make the return value `const` if it is a by-value return.